spec/repositories/answers_repository_spec.rb

```diff
@@ -30,4 +30,37 @@
       expect(session).to have_received(:dig).with(:dates)
     end
   end
+
+  describe "#clear_session - to allow a fresh application for landing" do
+    let(:session) do
+      {
+        "session_id" => "123",
+        "_csrf_token" => "ABC",
+        "dates" => {},
+        "destination" => {"destination_id" => "ABC123"},
+        "another_question" => "another answer"
+      }
+    end
```

Comment:
In terms of syntax, could you also define this in a more Javascripty way? i.e. 
  `{
        session_id: "123",
        _csrf_token: "ABC",
        dates: {},
        destination: {destination_id: "ABC123"},
        another_question: "another answer"
      }`

Reply:
Normally no, the keys in a normal `Hash` are either symbols or strings and are not interchangeable:

```rb
hash = {one: 1, "two" => 2}

> hash[:one]
=> 1

hash["one"]
=> nil
```

Rails' ActiveSupport library does offer a fancy hash which copes with both strings and symbols and which makes life easy when dealing with params in controllers:

```rb
fancy_hash = HashWithIndifferentAccess.new(one: 1, "two" => 2)

fancy_hash[:one]
> 1

fancy_hash["one"]
> 1
```

In this particular scenario we are dealing with `ActionDispatch::Request::Session` which does allow this flexible access, via string or symbol, but in it's own way: https://github.com/rails/rails/blob/5a0b2fa5a3be6ffd49fa7f1c3544c1da403c9cb5/actionpack/lib/action_dispatch/request/session.rb#L114